### PR TITLE
Purify

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib/ohmqtt/ohmqtt.cpp [hunk 1]
```diff
@@ -8,31 +8,42 @@
 ohmqtt::ohmqtt() {
   IPAddress broker(10,4,2,21);    // Address of the MQTT broker
```
Comment (ghost) [inline lib/ohmqtt/ohmqtt.cpp:9]: Is this always true or just for your instance?

File: lib/ohmqtt/ohmqtt.h [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
+#ifndef __OHMQTT_H
+#define __OHMQTT_H
 
 #include <ESP8266WiFi.h>
 #include <ESP8266mDNS.h>
 #include <PubSubClient.h>
 #include <ArduinoOTA.h>
-
-#ifdef __cplusplus
-extern "C" {
```
Comment (ghost) [inline lib/ohmqtt/ohmqtt.h:10]: This can not be used with ```class``` as class is not a keyword in C.

File: lib/ohmqtt/ohmqtt.h [hunk 3]
```diff
@@ -26,9 +23,9 @@ class ohmqtt {
   ESP8266WiFiClass* _WiFi;
   MQTT_CALLBACK_SIGNATURE;
   MQTT_RESUBSCRIBE_SIGNATURE;
-  const char* client_id;
```
Comment (ghost) [inline lib/ohmqtt/ohmqtt.h:29]: I removed const as this means it can only be set once, which were set to be null

File: src/main.cpp [hunk 4]
```diff
@@ -16,7 +16,7 @@ void resubscribe() {
 }
 
 void setup() {
-	Serial.begin(115200);
+  Serial.begin(115200);
```
Comment (ghost) [inline src/main.cpp:19]: These magic numbers bugs me

File: lib/ohmqtt/ohmqtt.cpp [hunk 5]
```diff
@@ -8,31 +8,42 @@
...
   WiFiClient wificlient;
-  this->client = new PubSubClient(wificlient);
-  this->broker = broker;
-  setCallback(NULL);
-  setResubscribe(NULL);
-  setClientID(NULL);
+  this->client      = new PubSubClient(wificlient);
+  this->broker      = broker;
+  this->callback    = nullptr;
```
Comment (ghost) [inline lib/ohmqtt/ohmqtt.cpp:13]: Setting all pointers to null within the class to ensure they are not dangling.

File: lib/ohmqtt/ohmqtt.h [hunk 6]
```diff
@@ -1,22 +1,19 @@
-#ifndef ohmqtt_h
-#define ohmqtt_h
+#ifndef __OHMQTT_H
```
Comment (ghost) [inline lib/ohmqtt/ohmqtt.h:1]: Making sure it wouldn't clash with any other symbol.

File: lib/ohmqtt/ohmqtt.cpp [hunk 7]
```diff
@@ -64,11 +75,26 @@ void ohmqtt::netSetup() {
   });
   ArduinoOTA.onError([](ota_error_t error) {
     Serial.printf("Error[%u]: ", error);
-    if      (error == OTA_AUTH_ERROR   ) Serial.println("Auth Failed");
-    else if (error == OTA_BEGIN_ERROR  ) Serial.println("Begin Failed");
-    else if (error == OTA_CONNECT_ERROR) Serial.println("Connect Failed");
-    else if (error == OTA_RECEIVE_ERROR) Serial.println("Receive Failed");
-    else if (error == OTA_END_ERROR    ) Serial.println("End Failed");
+    switch (error) {
```
Comment (ghost) [inline lib/ohmqtt/ohmqtt.cpp:78]: This looks gross but it is actually really neat as we save three instructions per if statement (which is noticeable when running code on IoT devices).

File: lib/ohmqtt/ohmqtt.cpp (hunk 1 above)
Comment (techman83) [reply lib/ohmqtt/ohmqtt.cpp:9]: Ideally I should be passing that in via a build flag. That was a future me problem to sort out.

File: lib/ohmqtt/ohmqtt.h (hunk 3 above)
Comment (techman83) [reply lib/ohmqtt/ohmqtt.h:29]: Ohh, that's why it was ending up null. Cool.

File: lib/ohmqtt/ohmqtt.cpp (hunk 7 above)
Comment (techman83) [reply lib/ohmqtt/ohmqtt.cpp:78]: Oh neat, I didn't know C++ had switch statements. Kinda gotten used to python not having them.

**Pull request merged**